Jenkinsfile

```diff
@@ -1,394 +1,303 @@
-                        failure_function(e, 'MacOSX / build+test failed')
-                    }
-                }
-
-            }
+        dir("${project}/build") {
+          try {
+            sh "cmake ../code"
+          } catch (e) {
+            failure_function(e, 'MacOSX / CMake failed')
+          }
+
+          try {
+            sh "make all UnitTests VERBOSE=1"
+            sh ". ./activate_run.sh && ./tests/UnitTests"
+          } catch (e) {
+            failure_function(e, 'MacOSX / build+test failed')
+          }
         }
+
+      }
     }
+  }
 }
 
 def get_system_tests_pipeline() {
-    return {
-        node('integration-test') {
-            cleanWs()
-            dir("${project}") {
-                try {
-                    stage("System tests: Checkout") {
-                        checkout scm
-                    }  // stage
-                    stage("System tests: Install requirements") {
-                        sh """scl enable rh-python35 -- python -m pip install --user --upgrade pip
-                        scl enable rh-python35 -- python -m pip install --user -r system-tests/requirements.txt
-                        """
-                    }  // stage
-                    stage("System tests: Run") {
-                        sh """cd system-tests/
-                        scl enable rh-python35 -- python -m pytest -s --junitxml=./SystemTestsOutput.xml .
-                        """
-                        junit "system-tests/SystemTestsOutput.xml"
-                    }  // stage
-                }finally {
-                    stage ("System tests: Clean Up") {
-                        sh """rm -rf system-tests/output-files/* || true
-                        docker stop \$(\$(docker ps -aq) | grep -E 'kafka|event-producer|zookeeper|filewriter|forwarder') || true
-                        docker rm \$(\$(docker ps -aq) | grep -E 'kafka|event-producer|zookeeper|filewriter|forwarder') || true
-                        """
-                    }  // stage
-                }
-            } // dir
-        }  // node
-    }  // return
+  return {
+    node('integration-test') {
+      cleanWs()
+      dir("${project}") {
+        try {
+          stage("System tests: Checkout") {
+            checkout scm
+          }  // stage
+          stage("System tests: Install requirements") {
+            sh """scl enable rh-python35 -- python -m pip install --user --upgrade pip
+            scl enable rh-python35 -- python -m pip install --user -r system-tests/requirements.txt
+            """
+          }  // stage
+          stage("System tests: Run") {
+            sh """cd system-tests/
+            scl enable rh-python35 -- python -m pytest -s --junitxml=./SystemTestsOutput.xml .
+            """
+            junit "system-tests/SystemTestsOutput.xml"
+          }  // stage
+        } finally {
+          stage ("System tests: Clean Up") {
+            sh """rm -rf system-tests/output-files/* || true
```

Comment:
This my fault for not doing it, but whilst you're here would you mind sticking a comment in here that says something like `the statements below return true because the build should pass even if there are no docker containers or output files to be removed` as it's probably not clear why there is a `|| true` on the end of the statements

Reply:
Sure, I'll add it.